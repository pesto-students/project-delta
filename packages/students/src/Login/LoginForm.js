import PropTypes from 'prop-types';
import React from 'react';

import BlockButton from '../../../shared-components/BlockButton';
import InputOutlined from '../../../shared-components/InputOutlined';

const LoginForm = ({
  onLogin, onEmailChange, email, isLoggingIn,
}) => (
  <form>
    <InputOutlined id="email" type="email" label="Email" onChange={onEmailChange} value={email} />
    <div className="login-button">
      <BlockButton type="submit" onClick={onLogin} isLoading={isLoggingIn}>Signin</BlockButton>
    </div>
  </form>
);

LoginForm.propTypes = {
  onLogin: PropTypes.func.isRequired,
  onEmailChange: PropTypes.func.isRequired,
  email: PropTypes.string.isRequired,
  isLoggingIn: PropTypes.bool.isRequired,
};

export default LoginForm;
