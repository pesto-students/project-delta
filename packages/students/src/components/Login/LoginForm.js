import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { BlockButton } from '../../../../shared-components/BlockButton';
import { InputOutlined } from '../../../../shared-components/InputOutlined';
import './style.css';

class LoginForm extends Component {
  state = {
    email: '',
  };

  onEmailChange = (event) => {
    this.setState({
      email: event.currentTarget.value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.handleLogin(this.state.email, this.clearEmail);
  };

  clearEmail = () => {
    this.setState({
      email: '',
    });
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} data-testid="loginForm">
        <InputOutlined
          data-testid="email"
          id="email"
          type="email"
          label="Email"
          onChange={this.onEmailChange}
          value={this.state.email}
        />
        <BlockButton className="login-button" type="submit" data-testid="signin" isLoading={this.props.isLoggingIn}>
          Signin
        </BlockButton>
      </form>
    );
  }
}

LoginForm.propTypes = {
  handleLogin: PropTypes.func.isRequired,
  isLoggingIn: PropTypes.bool.isRequired,
};

export { LoginForm };
