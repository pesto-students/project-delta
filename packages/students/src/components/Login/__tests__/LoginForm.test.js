import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { LoginForm } from '../LoginForm';

const updateInput = (wrapper, instance, newValue) => {
  const input = wrapper.find(instance);
  input.simulate('change', {
    currentTarget: { value: newValue },
  });
};

describe('<LoginForm />', () => {
  it('should allow user to enter email', () => {
    const wrapper = shallow(<LoginForm handleLogin={() => {}} isLoggingIn={false} />);

    updateInput(wrapper, '[data-testid="email"]', 'test@gmail.com');
    const emailInput = wrapper.find('[data-testid="email"]');

    expect(emailInput.props().value).toBe('test@gmail.com');
  });

  it('should allow user to submit the form', () => {
    const handleLogin = jest.fn();
    const wrapper = shallow(<LoginForm handleLogin={handleLogin} isLoggingIn={false} />);

    updateInput(wrapper, '[data-testid="email"]', 'test@gmail.com');
    wrapper.find('[data-testid="loginForm"]').simulate('submit', { preventDefault: () => {} });

    const instance = wrapper.instance();

    expect(handleLogin).toHaveBeenCalledWith('test@gmail.com', instance.clearEmail);
  });

  it('should reset email when clearEmail() called', () => {
    const handleLogin = jest.fn();
    const wrapper = shallow(<LoginForm handleLogin={handleLogin} isLoggingIn={false} />);
    updateInput(wrapper, '[data-testid="email"]', 'test@gmail.com');

    const instance = wrapper.instance();

    instance.clearEmail();

    expect(instance.state.email).toBe('');
  });

  test('matches saved snapshot', () => {
    const wrapper = shallow(<LoginForm handleLogin={() => {}} isLoggingIn={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
