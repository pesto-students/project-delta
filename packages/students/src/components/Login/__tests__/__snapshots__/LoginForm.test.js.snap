// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> matches saved snapshot 1`] = `
<form
  data-testid="loginForm"
  onSubmit={[Function]}
>
  <WithStyles(InputOutlinedComponent)
    data-testid="email"
    id="email"
    label="Email"
    onChange={[Function]}
    type="email"
    value=""
  />
  <div
    className="login-button"
  >
    <WithStyles(BlockButtonComponent)
      data-testid="signin"
      isLoading={false}
      type="submit"
    >
      Signin
    </WithStyles(BlockButtonComponent)>
  </div>
</form>
`;
